src/charm.py

```diff
@@ -175,29 +217,160 @@ def _generate_cu_config(self) -> str:
             cu_f1_interface_name=self._charm_config.f1_interface_name,
-            cu_f1_ip_address=pod_ip,
+            cu_f1_ip_address=str(self._charm_config.f1_ip_address).split("/")[0],
             cu_f1_port=self._charm_config.f1_port,
             du_f1_port=du_f1_port,
             cu_n2_interface_name=self._charm_config.n2_interface_name,
-            cu_n2_ip_address=pod_ip,
+            cu_n2_ip_address=str(self._charm_config.n2_ip_address).split("/")[0],
             cu_n3_interface_name=self._charm_config.n3_interface_name,
-            cu_n3_ip_address=pod_ip,
+            cu_n3_ip_address=str(self._charm_config.n3_ip_address).split("/")[0],
             amf_external_address=self._n2_requirer.amf_ip_address,
             mcc=self._charm_config.mcc,
             mnc=self._charm_config.mnc,
             sst=self._charm_config.sst,
             tac=self._charm_config.tac,
         )
 
+    def _generate_network_annotations(self) -> List[NetworkAnnotation]:
+        """Generate a list of NetworkAnnotations to be used by CU's StatefulSet.
+
+        Returns:
+            List[NetworkAnnotation]: List of NetworkAnnotations
+        """
+        return [
+            NetworkAnnotation(
+                name=f"{self._charm_config.n3_interface_name}-net",
+                interface=self._charm_config.n3_interface_name,
+            ),
+            NetworkAnnotation(
+                name=f"{self._charm_config.n2_interface_name}-net",
+                interface=self._charm_config.n2_interface_name,
+            ),
+            NetworkAnnotation(
+                name=f"{self._charm_config.f1_interface_name}-net",
+                interface=self._charm_config.f1_interface_name,
+            ),
+        ]
+
+    def _get_n3_nad_config(self) -> dict:
+        n3_nad_config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._charm_config.n3_ip_address,
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if self._charm_config.n3_subnet and self._charm_config.n3_gateway_ip:
+            n3_nad_config.update(
+                {
+                    "routes": [
+                        {
+                            "dst": str(self._charm_config.n3_subnet),
+                            "gw": str(self._charm_config.n3_gateway_ip),
+                        },
+                    ],
+                }
+            )
+        cni_type = self._charm_config.cni_type
+        if cni_type == CNIType.macvlan:
+            n3_nad_config.update(
+                {
+                    "type": "macvlan",
+                    "master": self._charm_config.n3_interface_name,
+                }
+            )
+        elif cni_type == CNIType.bridge:
+            n3_nad_config.update({"type": "bridge", "bridge": "n3-br"})
```

Comment:
This part looks like something we could create a separate method for. It repeats in every NAD config creation.
WDYT?

Reply:
These methods are reorganized by using a helper method to make the common things.